runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolLogUnit.java

```diff
@@ -0,0 +1,539 @@
+import org.corfudb.runtime.proto.service.LogUnit.WriteLogRequestMsg;
+import org.corfudb.runtime.proto.service.LogUnit.WriteLogResponseMsg;
+import org.corfudb.runtime.view.stream.StreamAddressSpace;
+
+import static org.corfudb.protocols.CorfuProtocolCommon.getStreamAddressSpaceMsg;
+import static org.corfudb.protocols.CorfuProtocolCommon.getTokenMsg;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUUID;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUuidMsg;
+import static org.corfudb.protocols.CorfuProtocolLogData.getLogData;
+import static org.corfudb.protocols.CorfuProtocolLogData.getLogDataMsg;
+import static org.corfudb.protocols.CorfuProtocolLogData.getReadResponseMsg;
+
+/**
+ * This class provides methods for creating the Protobuf objects defined
+ * in log_unit.proto. These provide the interface for performing the RPCs
+ * handled by the LogUnit server.
+ */
+@Slf4j
+public class CorfuProtocolLogUnit {
+    // Prevent class from being instantiated
+    private CorfuProtocolLogUnit() {}
+
+    /**
+     * Returns a READ request that can be sent by the client.
+     *
+     * @param addresses  a list of addresses to read from
+     * @param cacheable  true if the read result should be cached on the LogUnit server
+     * @return           a RequestPayloadMsg containing the READ request
+     */
+    public static RequestPayloadMsg getReadLogRequestMsg(List<Long> addresses, boolean cacheable) {
+        return RequestPayloadMsg.newBuilder()
+                .setReadLogRequest(ReadLogRequestMsg.newBuilder()
+                        .setCacheResults(cacheable)
+                        .addAllAddress(addresses)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Returns a READ response that can be sent by the server.
+     *
+     * @param addresses  a map containing the log data from the addresses read
+     * @return           a ResponsePayloadMsg containing the READ response
+     */
+    public static ResponsePayloadMsg getReadLogResponseMsg(Map<Long, LogData> addresses) {
+        return ResponsePayloadMsg.newBuilder()
+                .setReadLogResponse(ReadLogResponseMsg.newBuilder()
+                        .addAllResponse(addresses.entrySet()
+                                .stream()
+                                .map(e -> getReadResponseMsg(e.getKey(), e.getValue()))
+                                .collect(Collectors.toList()))
+                        .build())
+                .build();
+    }
+
+    /**
+     * Returns a ReadResponse from its Protobuf representation.
+     *
+     * @param msg  the desired Protobuf ReadLogResponse message
+     * @return     an equivalent ReadResponse object
+     */
+    public static ReadResponse getReadResponse(ReadLogResponseMsg msg) {
+        ReadResponse rr = new ReadResponse();
+        msg.getResponseList().forEach(e -> rr.put(e.getAddress(), getLogData(e.getLogData())));
+
+        return rr;
+    }
```

Comment:
Why does the response return another response? It should unwrap the resulted LogData and return that. No point in propagated the wrappers.